tests/test_directives.py

```diff
@@ -0,0 +1,266 @@
+                           'walltime': 20.0, 'memory': 16, 'processor_fraction': 0.5}
+        directives.update(valid_values_1)
+        for dirs in directives:
+            assert directives[dirs] == expected_values[dirs]
+
+    def test_update_directive_serial(self, available_directives_list, non_default_directive_values):
+        directives1 = Directives(available_directives_list)
+        directives2 = Directives(available_directives_list)
+        valid_values_0 = non_default_directive_values()
+        valid_values_1 = non_default_directive_values(1)
+        expected_values = {'np': 100, 'ngpu': 10, 'nranks': 5,
+                           'omp_num_threads': 20, 'executable': 'Non Default Path',
+                           'walltime': 84.0, 'memory': 32, 'processor_fraction': 0.5}
+        directives1.update(valid_values_0)
+        directives2.update(valid_values_1)
+        directives1.update(directives2, aggregate=True)
+        for dirs in directives1:
+            assert directives1[dirs] == expected_values[dirs]
+
+    def test_update_directive_parallel(
+        self, available_directives_list, non_default_directive_values
+    ):
+        directives1 = Directives(available_directives_list)
+        directives2 = Directives(available_directives_list)
+        valid_values_0 = non_default_directive_values()
+        valid_values_1 = non_default_directive_values(1)
+        expected_values = {'np': 104, 'ngpu': 11, 'nranks': 5,
+                           'omp_num_threads': 30, 'executable': 'Non Default Path',
+                           'walltime': 64.0, 'memory': 48, 'processor_fraction': 0.5}
+        directives1.update(valid_values_0)
+        directives2.update(valid_values_1)
+        directives1.update(directives2, aggregate=True, parallel=True)
+        for dirs in directives1:
+            assert directives1[dirs] == expected_values[dirs]
+
+    def test_evaluate_directive_none_job(
+        self, available_directives_list, non_default_directive_values
+    ):
+        directives = Directives(available_directives_list)
+        directives.evaluate(None)
+        valid_values = non_default_directive_values()
+        valid_values['processor_fraction'] = lambda job: job.sp.i/10
+        directives.update(valid_values)
+        with pytest.raises(RuntimeError):
+            directives.evaluate(None)
+
+    def test_evaluate_directive_valid_job(
+        self, available_directives_list, non_default_directive_values
+    ):
+        _tmp_dir = TemporaryDirectory(prefix='flow-directives_')
+        FlowProject.init_project(name='DirectivesTest',
+                                 root=_tmp_dir.name)
+        project = FlowProject.get_project(root=_tmp_dir.name)
+        for i in range(5):
+            project.open_job(dict(i=i)).init()
+
+        valid_values = non_default_directive_values()
+        valid_values['processor_fraction'] = lambda job: round(job.sp.i/10, 1)
```

Comment:
We are only testing `processor_fraction` we don't need the other non-default values.